src/core/hle/service/service.cpp

```diff
@@ -213,10 +213,16 @@ static bool AttemptLLE(const ServiceModuleInfo& service_module) {
 /// Initialize ServiceManager
 void Init(Core::System& core) {
     SM::ServiceManager::InstallInterfaces(core);
+    core.Kernel().SetAppMainThreadExtendedSleep(false);
 
     for (const auto& service_module : service_module_map) {
-        if (!AttemptLLE(service_module) && service_module.init_function != nullptr)
+        const bool has_lle = AttemptLLE(service_module);
+        if (!has_lle && service_module.init_function != nullptr) {
             service_module.init_function(core);
+        }
+        if (has_lle) {
+            core.Kernel().SetAppMainThreadExtendedSleep(true);
+        }
```

Comment:
```suggestion
        core.Kernel().SetAppMainThreadExtendedSleep(has_lle);
```
Prob faster to avoid the branch since its a simple variable assignment

Reply:
This is not the same behaviour, if a later iteration of the for loop results in has_lle = false, it will be set back from true to false.